### PR TITLE
Fixes PCI validation to use decimal format

### PR DESCRIPTION
**Pull request by srm09**
### What this PR does / why we need it
As, CAPV expects the PCI inputs of deviceID:vendorID in a Decimal format, this patch fixes the validation to compare against the Decimal values of the Tesla T4 Hex values. 
It also sets the deviceID vendorID values in the map as an integer instead of a string.

### Which issue(s) this PR fixes
Fixes #

### Describe testing done for PR
- Created a cluster with GPU enabled worker nodes via PCI passthrough in a vSphere testbed.

### Release note
```release-note
NONE
```

### CODE REVIEW

**Comment by jayunit100** [comment]
approveeeddd

**Review by jayunit100 — APPROVED** [review]

**Review by prkalle — APPROVED** [review]
LGTM. Thanks!

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml [hunk 1]
```diff
@@ -69,6 +38,41 @@
...
 #@ end
 
+
+#@ valid_pci_devices = [[0x10DE,0x1EB8]]
+
+#@ def valid_pci_devices_list():
+#@       return ",".join(map(lambda x:  "0x%X:0x%X" % (x[0], x[1]), valid_pci_devices))
+#@ end
+
+#@ def get_pci_devices(pci_devices_string, pci_ignore_pci_device_validation):
```
Comment (sathyanarays) [inline pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:48]: NIT: The variable name could be either pci_ignore_device_validation or ignore_pci_device_validation.

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml [hunk 2]
```diff
@@ -69,6 +38,41 @@
...
+#@         if len(val) <= 0:
+#@             assert.fail(help_error)
+#@         end
+#@         kv = val.split(":")
+#@         if len(kv) != 2:
+#@             assert.fail(help_error)
+#@         end
+#@         id_pair = dict()
+#@         vendor_id = int(kv[0], 16)
+#@         device_id = int(kv[1], 16)
```
Comment (sathyanarays) [inline pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:61]: Should we return specific error message if the deviceId or vendorId provided is not a valid hexadecimal?

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml [hunk 3]
```diff
@@ -69,6 +38,41 @@
...
+#@         if not pci_ignore_pci_device_validation:
+#@             for x in valid_pci_devices:
+#@                 if vendor_id != x[0] or device_id != x[1]:
+#@                     assert.fail("Device 0x%X:0x%X is not supported. Supported devices are: %s. Set VSPHERE_IGNORE_PCI_DEVICES_ALLOW_LIST to ignore this check." % (vendor_id, device_id, valid_pci_devices_list()))
+#@                 end
+#@             end
+#@         end
```
Comment (sathyanarays) [inline pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:68]: Shouldn't the logic be like this?

```
#@        if not pci_ignore_pci_device_validation:
#@.            matchesAny = false
#@             for x in valid_pci_devices:
#@                 if vendor_id == x[0] or device_id == x[1]:
#@                     matchesAny = true
#@                 end
#@             end
#@             if not matchesAny:
#@                 assert.fail("Device 0x%X:0x%X is not supported. Supported devices are: %s. Set VSPHERE_IGNORE_PCI_DEVICES_ALLOW_LIST to ignore this check." % (vendor_id, device_id, valid_pci_devices_list()))
#@             end
#@         end
```

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml (hunk 2 above)
Comment (randomvariable) [reply pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:61]: There's no try-catch support for this in Starlark unfortunately.

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml (hunk 3 above)
Comment (randomvariable) [reply pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:75]: Apart from `if vendor_id == x[0] or device_id == x[1]:`
needing to be `if vendor_id == x[0] and device_id == x[1]:`, and for which i've now added a test case just now, yup. Changed.

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml (hunk 1 above)
Comment (randomvariable) [reply pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml]: Done

File: pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml (hunk 2 above)
Comment (randomvariable) [reply pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml:64]: Added a comment that this needs to be covered in documentation.

**Review by sathyanarays — APPROVED** [review]

**Review by yastij — APPROVED** [review]

**Pull request merged**